lib/fix_ints.nit

```diff
@@ -0,0 +1,789 @@
+		return (void*)(intptr_t)s;
+	`}
+
+	redef fun min_int do return -128
+
+	redef fun max_int do return 127
+
+	redef fun to_i8 `{ return self;`}
+
+	redef fun to_i16 `{ return (void*)(intptr_t)((int16_t)((int8_t)(intptr_t)self));`}
+
+	redef fun to_u16 `{ return (void*)(intptr_t)((uint16_t)((int8_t)(intptr_t)self));`}
+
+	redef fun to_i32 `{ return (void*)(intptr_t)((int32_t)((int8_t)(intptr_t)self));`}
+
+	redef fun to_u32 `{ return (void*)(intptr_t)((uint32_t)((int8_t)(intptr_t)self));`}
+
+	redef fun to_i `{ return (long)(int8_t)(intptr_t)self;`}
+
+	redef fun to_b `{ return (unsigned char)(int8_t)(intptr_t)self;`}
+
+	redef fun +(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv + rv);
+		return retval;
+	`}
+
+	redef fun -(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv - rv);
+		return retval;
+	`}
+
+	redef fun *(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv * rv);
+		return retval;
+	`}
+
+	redef fun /(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv / rv);
+		return retval;
+	`}
+
+	redef fun %(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv % rv);
+		return retval;
+	`}
+
+	redef fun >>(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv >> rv);
+		return retval;
+	`}
+
+	redef fun <<(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv << rv);
+		return retval;
+	`}
+
+	redef fun |(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv | rv);
+		return retval;
+	`}
+
+	redef fun &(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv & rv);
+		return retval;
+	`}
+
+	redef fun ~ `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		void* retval = (void*)(intptr_t)(~lv);
+		return retval;
+	`}
+
+	redef fun ^(o) `{
+		int8_t lv = (int8_t)(intptr_t)self;
+		int8_t rv = (int8_t)(intptr_t)o;
+		void* retval = (void*)(intptr_t)(lv ^ rv);
+		return retval;
+	`}
+
+	# C function to calculate the length of the `NativeString` to receive `self`
+	redef fun strlen `{
+		int8_t s = (int8_t)(intptr_t)self;
+		return snprintf(NULL, 0, "%hhd", s);
+	`}
+
+	# C function to convert an nit Int16 to a NativeString (char*)
+	redef fun native_to_s(nstr, strlen) `{
+		int8_t s = (int8_t)(intptr_t)self;
+		snprintf(nstr, strlen, "%hhd", s);
+	`}
+
+	redef fun output `{ int8_t s = (int8_t)(intptr_t)self; printf("%hhd", s);`}
+end
+
+# Signed integer with 16-bit of precision
+#
+# Max int = +32 767
+# Min int = -32 768
+extern class Int16 `{ int16_t * `}
```

Comment:
From what I can gather, the precise pointer `int16_t*` is no different than a `void*`. Only the address data of the pointer is used.